File: backend/app/schemas/agent_run.py
"""Schemas for the Agent Ops surface: live agent state + run history."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import AgentName, AgentRunStatus


class AgentRunItem(BaseModel):
    """One row of agent activity — a single agent invocation, start to finish."""

    model_config = ConfigDict(from_attributes=True)

    id: str
    agent_name: AgentName
    status: AgentRunStatus
    started_at: datetime
    finished_at: datetime | None = None
    input_summary: str | None = None
    output_summary: str | None = None
    error: str | None = None
    linked_entity_type: str | None = None
    linked_entity_id: str | None = None


class AgentNodeState(BaseModel):
    """The live-graph view's per-agent node: its most recent run, whatever that is."""

    agent_name: AgentName
    status: AgentRunStatus | None = None
    last_run: AgentRunItem | None = None


class AgentRunListResponse(BaseModel):
    """Filtered run history plus the current node state for every named agent."""

    items: list[AgentRunItem]
    total: int
    nodes: list[AgentNodeState]

File: frontend/src/types/agentRun.ts
/** The five named agents (`app.models.enums.AgentName`), in the fixed hand-off order the
 *  graph renders them: sourcing, then eligibility, then scoring, then applying, then tracking. */
export type AgentName = 'discovery' | 'eligibility' | 'scoring' | 'application' | 'tracking';

export type AgentRunStatus = 'running' | 'done' | 'error' | 'needs_review';

export interface AgentRunItem {
  id: string;
  agent_name: AgentName;
  status: AgentRunStatus;
  started_at: string;
  finished_at: string | null;
  input_summary: string | null;
  output_summary: string | null;
  error: string | null;
  linked_entity_type: string | null;
  linked_entity_id: string | null;
}

export interface AgentNodeState {
  agent_name: AgentName;
  status: AgentRunStatus | null;
  last_run: AgentRunItem | null;
}

export interface AgentRunListResponse {
  items: AgentRunItem[];
  total: number;
  nodes: AgentNodeState[];
}

export interface AgentRunFilters {
  agent_name?: AgentName;
  status?: AgentRunStatus;
  linked_entity_type?: string;
  linked_entity_id?: string;
  page?: number;
  page_size?: number;
}
